src/com/github/korniloval/matlab/editor/smartEnter/StatementCompletionProcessor.kt

```diff
@@ -0,0 +1,64 @@
+package com.github.korniloval.matlab.editor.smartEnter
+
+import com.github.korniloval.matlab.psi.MatlabAssignExpr
+import com.github.korniloval.matlab.psi.MatlabFunctionExpr
+import com.intellij.codeInsight.editorActions.smartEnter.SmartEnterProcessor
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.util.TextRange
+import com.intellij.openapi.util.text.StringUtil
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiFile
+import java.util.*
+import java.util.Optional.*
+
+class StatementCompletionProcessor : SmartEnterProcessor() {
+
+  override fun process(project: Project, editor: Editor, psiFile: PsiFile): Boolean {
+    return ofNullable(getStatementAtCaret(editor, psiFile))
+        .flatMap { element ->
+          findStatement(element) { node ->
+            when (node) {
+              is MatlabFunctionExpr,
+              is MatlabAssignExpr -> true
+              else -> false
+            }
+          }
+        }
+        .map { statementElement ->
+          val statementRange = getStatementRange(editor, statementElement)
+          val fullStatement = editor.document.getText(statementRange)
+          val endOfStatement =
+              completeStatementIfNecessary(fullStatement, editor, statementElement.textRange.endOffset)
+          editor.caretModel.moveToOffset(endOfStatement + 1)
+          true
+        }.orElse(true)
+
+  }
+
+  private fun getStatementRange(editor: Editor, statementElement: PsiElement): TextRange =
+      if (editor.document.textLength <= statementElement.textRange.endOffset) {
+        statementElement.textRange
+      } else {
+        statementElement.textRange.grown(1)
+      }
+
+  private fun completeStatementIfNecessary(fullStatement: String,
+                                           editor: Editor,
+                                           endOfStatement: Int): Int =
```

Comment:
maybe it's a good idea to change return value to `Int?` and return `null` when we don't need to complete a statement (so we can `return false` from `process` in this case)